aten/src/ATen/cuda/CUDAGeneratorImpl.cpp

```diff
@@ -47,6 +54,8 @@ const Generator& getDefaultCUDAGenerator(DeviceIndex device_index) {
   std::call_once(cuda_gens_init_flag[idx], [&] {
     default_gens_cuda[idx] = make_generator<CUDAGeneratorImpl>(idx);
     default_gens_cuda[idx].seed();
+    // chooses minor ugliness over adding an arg to make_generator
+    reinterpret_cast<CUDAGeneratorImpl*>(&default_gens_cuda[idx])->set_is_default(true);
```

Comment:
OK.

Reply:
Got rid of set_is_default, using pointer checks instead.